### PR TITLE
Feature/15104 today widget data

### PR DESCRIPTION
**Pull request by Gio2018**
Ref #15104

This PR only prepares the `WordPressHomeWidgetToday` target to use the existing types used for the old Today widget. Also removes unnecessary additional entitlements file.

It does not (yet):
- Refresh data after the first load
- Allow to configure the site
- request fresh data from the backend
- Separate data from the old Today Widget

To test:

- Build run and install either size of the new Today widget
- You should see Today data from whatever site you had set for the old widgets (see example below)

<p align=center>

</p>

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by frosty — APPROVED** [review]

**Comment by ScoutHarris** [comment]
Hey @Gio2018 .

This is something that can be addressed later, but if a site has no title nothing is displayed. You may want to discuss with @mattmiklic , but maybe we could show the URL if there is no title? Pretty sure we do that in other places in the app.

no_site_title

File: WordPress/WordPressHomeWidgetToday/Views/TodayWidgetMediumView.swift [hunk 1]
```diff
@@ -16,15 +16,15 @@
...
-                           lowerValue: "\(content.likes)")
+                           lowerValue: "\(content.stats.likes)")
                 Spacer()
                 Spacer()
                 makeColumn(upperTitle: visitorsTitle,
-                           upperValue: "\(content.visitors)",
+                           upperValue: "\(content.stats.visitors)",
                            lowerTitle: commentsTitle,
-                           lowerValue: "\(content.comments)")
+                           lowerValue: "\(content.stats.comments)")
```
Comment (ScoutHarris) [inline WordPress/WordPressHomeWidgetToday/Views/TodayWidgetMediumView.swift]: Just a reminder - all the stats values need `.abbreviatedString()` appended. You can either do this now or later, up to you.

File: WordPress/WordPressHomeWidgetToday/Views/TodayWidgetSmallView.swift [hunk 2]
```diff
@@ -11,7 +11,7 @@ struct TodayWidgetSmallView: View {
                 FlexibleCard(axis: .vertical, title: widgetTitle, value: content.siteTitle)
 
                 Spacer()
-                VerticalCard(title: viewsTitle, value: "\(content.views)", largeText: true)
+                VerticalCard(title: viewsTitle, value: "\(content.stats.views)", largeText: true)
```
Comment (ScoutHarris) [inline WordPress/WordPressHomeWidgetToday/Views/TodayWidgetSmallView.swift]: Same here - `.abbreviatedString()` is needed.

File: WordPress/WordPressHomeWidgetToday/Views/TodayWidgetView.swift [hunk 3]
```diff
@@ -44,6 +44,9 @@ extension TodayWidgetView {
 
 struct TodayWidgetView_Previews: PreviewProvider {
     static var previews: some View {
+        TodayWidgetView(content: staticModel)
+            .previewContext(WidgetPreviewContext(family: .systemSmall))
+
```
Comment (ScoutHarris) [inline WordPress/WordPressHomeWidgetToday/Views/TodayWidgetView.swift:49]: 👍

**Review by ScoutHarris — APPROVED** [review]
Just a couple suggestions. But otherwise, :shipit: !

**Comment by mattmiklic** [comment]
> maybe we could show the URL if there is no title?

Good catch, and good idea @ScoutHarris. I think this is what we should do.

File: WordPress/WordPressHomeWidgetToday/Views/TodayWidgetMediumView.swift (hunk 1 above)
Comment (Gio2018) [reply WordPress/WordPressHomeWidgetToday/Views/TodayWidgetMediumView.swift]: Thanks for catching this!! Fixed

File: WordPress/WordPressHomeWidgetToday/Views/TodayWidgetSmallView.swift (hunk 2 above)
Comment (Gio2018) [reply WordPress/WordPressHomeWidgetToday/Views/TodayWidgetSmallView.swift]: Fixed

**Comment by Gio2018** [comment]
Thanks for reviewing @frosty @ScoutHarris @mattmiklic !!

the blank site name should be fixed now, using the site url in case it's empty

File: WordPress/WordPressHomeWidgetToday/WordPressHomeWidgetToday.swift [hunk 4]
```diff
@@ -30,9 +30,18 @@ struct Provider: TimelineProvider {
...
-        UserDefaults(suiteName: WPAppGroupName)?.string(forKey: WPStatsTodayWidgetUserDefaultsSiteNameKey) ?? "Site name not found"
+        let defaults = UserDefaults(suiteName: WPAppGroupName)
+
+        if let title = defaults?.string(forKey: WPStatsTodayWidgetUserDefaultsSiteNameKey),
+           !title.isEmpty {
+            return title
+        } else if let url = defaults?.string(forKey: WPStatsTodayWidgetUserDefaultsSiteUrlKey),
+                  !url.isEmpty {
+            return url
+        } else {
```
Comment (ScoutHarris) [inline WordPress/WordPressHomeWidgetToday/WordPressHomeWidgetToday.swift]: Nitpicky, but these `else`s aren't necessary since each case `return`s.

Comment (Gio2018) [reply WordPress/WordPressHomeWidgetToday/WordPressHomeWidgetToday.swift]: Yup! Done

**Comment by ScoutHarris** [comment]
> the blank site name should be fixed now, using the site url in case it's empty

Confirmed. Thanks!

**Pull request merged**